src/electron-main.ts

```diff
@@ -192,11 +191,13 @@ async function setupGlobals(): Promise<void> {
     }
 
     // The tray icon
-    // It's important to call `path.join` so we don't end up with the packaged asar in the final path.
-    const iconFile = `element.${process.platform === 'win32' ? 'ico' : 'png'}`;
-    iconPath = path.join(resPath, "img", iconFile);
+    // It's important to call `path.join` for the bundled assets so we don't end up with the packaged asar in the final path.
+    const icon_path = process.platform === 'win32'
+      ? (global.vectorConfig?.tray_icons.ico || path.join(resPath, 'img', 'element.ico'))
+      : (global.vectorConfig?.tray_icons.png || path.join(resPath, 'img', 'element.png'));
```

Comment:
After thinking about it a bit more, I don't think this is the right approach. This is a user config-- I don't think it makes sense to constrain it to particular file types.

In your first version, that just had a single path,  @t3chguy said
> As per Electron docs, ICO is recommended on Windows, this configuration model doesn't support this.

How does a simple path not support ico? The user can just provide an ico path if they're configuring this on windows. The config option can be documented to inform the user what kind of icon they should use for their system. Why would a user, who's using `element-desktop` on a particular system, provide configs for multiple system tray file formats?

I'd recommend reverting to just specifying a single path, not different formats, if @t3chguy agrees with my assessment.

Or am I misunderstanding that this is an end user config? Is this intended for packagers to be able to provide custom icons? I just got element set up the other day so I'm quite possibly missing context!

Reply:
also tagging @turt2live since they've commented on providing an ico as well.

Reply:
The config here is a *build* config, not a *user* config, ftr.

I'm not concerned with the API shape in all honesty: it just needs to work and not look awful ;)

Reply:
Ah, thank you, I completely revoke my comment! Makes sense.

How would people feel about me adding a user config in a separate PR to change the tray icon? I'm using a custom icon pack on Linux, and I'd like to have the tray icon match.

Reply:
That would require a product decision, though I don't foresee it being a successful change proposal if I'm being honest ;) 

Rationale being that things like brand config are meant to be build-level things and not customizable by the user, even if technically possible.

Reply:
`config.json` is is still exposed and documented as user run-time configuration and I really hope that this will continue being supported, regardless how commonly it might be used today.

https://github.com/vector-im/element-desktop/blob/cc94bbea14e6a0bd4bbef72b8ad969b05913074e/README.md#user-specified-configjson

Reply:
it's supported as a user config as well, but primarily this feature would be used as part of a build pipeline - we would be unlikely to support it as part of a user-supplied config (if an issue were to be opened as such).